unit-tests/src/test/java/org/eclipse/collections/impl/factory/SetsTest.java

```diff
@@ -762,6 +762,21 @@ public void cartesianProduct()
     }
 
     @Test
+    public void cartesianProductWithFunction()
+    {
+        MutableSet<Integer> set1 = UnifiedSet.newSetWith(1, 2);
```

Comment:
Eventually, yes.  I held off doing this for now as it is a lot of work, and would make a much larger review.  I can submit one or two more pull requests implementing the new cartesianProduct on our Set interfaces.  This can include the primitive interfaces as well now that we have implementations available on IntSets.